nexus/db-queries/src/db/datastore/mod.rs

```diff
@@ -276,6 +291,11 @@ impl DataStore {
         Ok(datastore)
     }
 
+    /// Terminates the underlying pool, stopping it from connecting to backends.
+    pub async fn terminate(&self) {
+        self.pool.terminate().await
+    }
```

Comment:
How long do we generally expect this to take? Wondering if we need to have a timeout here (or maybe there's a timeout elsewhere? If so a comment would be helpful.)

Reply:
It *should* be really quick. This fires a oneshot, which causes all tokio tasks to terminate ASAP. I'm using `biased` select loops, and prioritizing early exiting if this is fired.